app/models/vendor_code.rb

```diff
@@ -7,11 +7,8 @@ class VendorCode < ActiveRecord::Base
                    length: { maximum: 255, message: "longer than 255 characters" }
 	
 	
-  def assign_to(current_user)
-    self.update_attributes(user: current_user, user_name: current_user.name, email: current_user.email)
-    vendor = self.vendor
-    vendor.update_attribute(:usedCodes, vendor.usedCodes + 1)
-    vendor.update_attribute(:unclaimCodes, vendor.unclaimCodes - 1)
+  def associate_with(user)
+    self.update(user_id: user.id, user_name: user.name, email: user.email)
   end  
```

Comment:
@alexander-kh:
Hi Alex,
any particular reason to remove the adjustments of the usedCodes and  unclaimCodes attributes when a vendor code gets associated with a user ? Also, the related respec example, i.e. 'updates vendor usedCodes and unclaimCodes attributes', testing such updates in vendor_code_spec.rb, for some reason, has been removed. 
Unless the attribute updates have been refactored to a separate function (yet unknown to me), this would seem like a bug or there is a change in initial requirements.